### PR TITLE
Netstandard fixes for the logging mechanism

### LINKED ISSUE #2541: Remove usage of MvxTrace from code
### Steps to reproduce :scroll:
Build MvvmCross.sln

### Expected behavior :thinking:
No warnings

### Actual behavior :bug:
Over 200 warnings about MvxTrace and derivatives

Since we have deprecated the static methods in favor of the new IMvxLog stuff, we are not getting a whole bunch of warnings when we build. We should migrate our own code.

### PR DESCRIPTION
**Pull request by willsb**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Improvements

### :arrow_heading_down: What is the current behavior?
The logger heavily relies on reflection for basic logging due to PCL constraints and MvxTrace is there, marked as obsolete

### :new: What is the new behavior (if this is a feature change)?
We leverage netstandard to remove reflection from the basic providers and burninate MvxTrace and the likes of it.

### :boom: Does this PR introduce a breaking change?
Yessir

### :bug: Recommendations for testing
As discussed with @martijn00 and @nmilcoff, this was developed under VS4Mac, which has a very limited support for netstandard. I very much recommend checking this on Windows for completeness sake.

### :memo: Links to relevant issues/docs
Closes #2541
A rework of #2399

### :thinking: Checklist before submitting

- [ ] All projects build (This needs to be checked. I removed the bulk of things, but VS4Mac is limited and I can't check that usages of the MvxTrace were properly removed from Plugins and other projects)
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [X] Nuspec files were updated (when applicable)
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by Cheesebaron** [comment]
Doesn't build at all. MvxTrace is still used all over the place.

File: MvvmCross.Android.Support/Core.UI/MvxAndroidLog.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿// MvxBindingLog.cs
```
Comment (Cheesebaron) [inline MvvmCross.Android.Support/Core.UI/MvxAndroidLog.cs]: Not sure I am a big fan of these duplicated Log classes.

Comment (martijn00) [reply MvvmCross.Android.Support/Core.UI/MvxAndroidLog.cs]: We could have one file and link to that in each project, but not sure if that would make it any better. What i want to do is change the log name per project so the logging output makes more sense.

Comment (Cheesebaron) [reply MvvmCross.Android.Support/Core.UI/MvxAndroidLog.cs]: Right. Make a new issue which describes this, then we can merge this one so it isn't blocking the rest of the 6.0.0 stuff.

**Pull request merged**